mapper-processor/src/test/java/com/datastax/oss/driver/internal/mapper/processor/dao/DaoImplementationGeneratorTest.java

```diff
@@ -23,6 +23,7 @@
 
 public class DaoImplementationGeneratorTest extends DaoMethodGeneratorTest {
 
+  // todo handle Update where there is no set
```

Comment:
fails in this test are related to this line:
https://github.com/datastax/java-driver/pull/1235/files#diff-75706beabe9b72dbf270c073d07d201fR61
To set all columns for Update we need to get all non PKs columns: `entityDefinition.getRegularColumns`.
If there is a case (like in this test) that the Entity for which we are generating update prepared statement has only one column and that column is PK the generated Update statement is incorrect.
(The WHERE clause in Update is mandatory and it cannot contains `PK`). Because of that, the generated update statement will be incorrect. I am not sure how should we handle such a situation: maybe we should not generate update code path if the entity has only PK? But if the user is annotating the method with `@Update` for such entity it will be confusing for end-user: so other solution will be to throw an exception that we cannot generate update for the entity with only one column. Wdyt?

To make this test pass I can just add one additional non-PK key to `protected static final TypeSpec ENTITY_SPEC =` in the DaoMethodGeneratorTest but I am not 100% sure why in this test the Mapper tries to auto-generate  
```
@Override
  public UpdateWithAssignments update() {
```

Reply:
Yes, I think we should throw in corner cases like this. I handled a similar case in `EntityHelperDeleteByPrimaryKeyMethodGenerator`, you can see the generated `throw` statement at the beginning of the `generate()` method.

Reply:
Error handling implemented in https://github.com/datastax/java-driver/pull/1235/commits/07644279cf79e5e86905da796da7d75c14badb1b.